### ISSUE TITLE
lack "Category-consistency Contrastive Loss Formulation"

### ISSUE THREAD

**Issue by GUESSZERO4** [LYH-hh/AFiRe#1]
Hello, and thank you for open‑sourcing AFiRe: Anatomy‑Driven Self‑Supervised Learning for Fine‑Grained Representation in Radiographic Images. I have been studying the loss implementation and noticed the following:

In the paper (Eqs. 12 & 16), the token‐wise contrastive loss is composed of two terms: structure‑consistency (L<sub>stru.cst</sub>) and category‑consistency (L<sub>cate.cst</sub>).

In the provided AFiReLoss class, the forward method computes the structure‑consistency loss (cst_loss) and the pixel‑level reconstruction loss (recon_loss), but I was unable to find where the category‑consistency term (L<sub>cate.cst</sub>) is implemented.

Could you please clarify:

Whether L<sub>cate.cst</sub> is intentionally omitted or simplified in this implementation?

If it is implemented elsewhere, could you point me to the relevant code section or branch?

If it is not yet implemented, do you plan to add it in a future update?

Thank you very much for your time and for sharing this excellent work. I look forward to your guidance.

**Comment by LYH-hh** [comment]
@GUESSZERO4  
### Implementation Details of Category-Consistency Loss

1. ** Category Token Selection**:
   - Category-related tokens are selected using:
     ```python
     indices_normal = torch.nonzero(patches_labels == 0, as_tuple=False)  # indices of shape (B,L)
     ```
   - This identifies tokens corresponding to normal anatomical structures.

2. **Loss Computation**:
   - The category-consistency term is jointly implemented with the structure-consistency term in the following loop:
     ```python
     for (i,j) in indices_normal:
         loss = torch.sum(-teacher_out[j] * F.log_softmax(student_out[i,j], dim=-1), dim=-1)
         cst_loss += loss.mean()
         n_loss_terms += 1
     ```
   - Since all positive queries (Q⁺) are already aligned with the prototypes (C) representing normal anatomical structures, we only need to additionally constrain the negative queries (Q⁻) to stay away from these prototypes.
   - This approach ensures that:
     - Normal patterns maintain consistency with the established prototypes
     - Abnormal patterns are clearly distinguished from normal prototypes
